htlcswitch/switch.go

```diff
@@ -25,6 +26,11 @@ var (
 	zeroPreimage [sha256.Size]byte
 )
 
+// To ensure we never accidentally cause an HTLC overflow, we'll use
+// this buffered channel as as semaphore in order to limit the number
+// of outstanding HTLC's we extend to the target link.
+const numSlots = lnwallet.MaxHTLCNumber - 5
```

Comment:
It's worth noting that the `MaxHTLCNUmber` actually applies to the max number of HTLC's _each_ side is allowed to have unsettled within a given commitment transaction. So this value should actually be using _half_ of the constant exposed from the lnwallet package.